src/test/server.c

```diff
@@ -36,7 +36,12 @@ static int vdisk_read(struct qnio_msg *msg, struct iovec *returnd)
     size = msg->hinfo.io_size;
     safe_strncpy(vdisk_path_temp, msg->hinfo.target, DIR_NAME_SZ);
     bname = basename(vdisk_path_temp);
-    sprintf(vdisk_path, "%s/%s", vdisk_dir, bname);
+    if (strlen(bname) + strlen(vdisk_dir) > DIR_NAME_SZ - 1) {
+        fprintf(stderr, "Combined length of directory path and "
+                "filename exceeds %d.\n", DIR_NAME_SZ - 1);
+        exit(-1);
```

Comment:
POSIX says "only the least significant 8 bits (that is, status & 0377) shall be available to a waiting parent process":
http://pubs.opengroup.org/onlinepubs/000095399/functions/exit.html

Error status codes are positive.  Please use EXIT_FAILURE instead of -1.